### ISSUE TITLE
[Flax] Add FlaxBart model

### ISSUE THREAD

**Issue by stancld** [huggingface/transformers#11478]
# 🚀 Feature request

It would be nice to implement Flax version of BART model.

## Motivation

Narrow the gap in support between encoder transformers (BERT, RoBERTa, ...) and encoder-decoder models (BART,...).

## Your contribution

I've been working on this now so I hope I will send a PR soon.

@patrickvonplaten @sgugger

**Comment by patrickvonplaten** [comment]
Wow this sounds awesome! Let me know if you need help! (I think we will need to look at the `generate()` function) together :-)

**Comment by stancld** [comment]
Hi @patrickvonplaten, as I indicated I've started working on `FlaxBart` which can be found on this branch https://github.com/stancld/transformers/tree/FlaxBart . So far, I've implemented `FlaxBartModel` and `FlaxBartForConditionalGeneration` with some remaining pieces to do, but it is possible to run them. 

As there is no official template for Flax encoder-decoder models, I've tried to follow Torch implementation of Bart and Flax implementation of Bert. Before diving deeper and finishing all the components, tests etc, could I, please, ask you to provide me with short feedback if this structure seems ok to you? Thanks a lot in advance! :)

**Comment by patrickvonplaten** [comment]
This sounds great :-) Thanks a lot for tackling this! Could you maybe make a [WIP] PR from your branch and ping me - this would make it a bit easier to review the code

**Comment by patrickvonplaten** [comment]
Hey @stancld, 

I looked quickly and in general the PR already looks great :-) 

A couple of things:
- we don't allow `labels` as an input argument to Flax models (and actually probably even won't do this in the future). Flax/Jax is inherently functional which means that a loss function should wrap the model forward function and not the other way around
- Weight tying is done a bit differently as in PyTorch, thus we don't need `    def get_input_embeddings(self):` for now
- `return_dict` is now also implemented in `FlaxBert` -> so this can be copied from there :-) 

=> In short the design already looks great :-) I think you can open a PR & we'll discuss everything on the PR

**Comment by stancld** [comment]
@patrickvonplaten Thanks a lot for the feedback! I will create [WIP] PR later today :)

**Comment by patrickvonplaten** [comment]
ping
